### DISCUSSION TITLE
Proposal: Basic Digital Asset Management (DAM) Enhancements in Umbraco

### DISCUSSION THREAD

**Discussion by Girish-kl33** [umbraco/Umbraco-CMS#17538] [Features and ideas]
<p>Currently, Umbraco relies on third-party tools for advanced Digital Asset Management (DAM). While these integrations are useful, adding some basic DAM features directly within Umbraco's core functionality would greatly improve the user experience for content editors and simplify asset management. Below are the proposed enhancements:</p>
<hr>
<h3><strong>Proposed Enhancements</strong></h3>
<ol>
<li>
<p><strong>Image License Field</strong></p>
<ul>
<li>A field to specify licensing details for each image (e.g., "Creative Commons," "Royalty-Free").</li>
<li>This will help editors manage legal and compliance aspects directly within Umbraco.</li>
</ul>
</li>
<li>
<p><strong>Fallback Image Configuration</strong></p>
<ul>
<li>Allow users to set a fallback image for each asset.</li>
<li>If the primary image is no longer valid (e.g., expired or unavailable), the fallback image will be displayed automatically.</li>
</ul>
</li>
<li>
<p><strong>Expiration Date Field</strong></p>
<ul>
<li>Add a date picker field for assets to define the "valid until" date.</li>
<li>Once the expiration date is reached, the system will automatically display the fallback image.</li>
</ul>
</li>
</ol>
<hr>
<h3><strong>How It Works</strong></h3>
<ul>
<li><strong>In the Media Section</strong>:
<ul>
<li>Extend the default media types to include:
<ul>
<li>"Image License" (Textstring/Dropdown)</li>
<li>"Fallback Image" (Media Picker)</li>
<li>"Expiration Date" (Date Picker)</li>
</ul>
</li>
</ul>
</li>
<li><strong>Template Logic</strong>:
<ul>
<li>Add simple logic to templates to check if the expiration date has passed. If expired, render the fallback image.</li>
<li>Example:
<div class="contain-inline-size rounded-md border-[0.5px] border-token-border-medium relative bg-token-sidebar-surface-primary dark:bg-gray-950">
<div class="overflow-y-auto p-4" dir="ltr">@if (Model.Media.ExpirationDate != null &amp;&amp; Model.Media.ExpirationDate &lt; DateTime.Now) <br>{<br>&nbsp; &nbsp; &nbsp; &nbsp;&lt;img src="@Umbraco.Media(Model.Media.FallbackImage).Url" alt="Fallback Image" /&gt; <br>} <br>else <br>{ <br>&nbsp; &nbsp; &nbsp; &lt;img src="@Umbraco.Media(Model.Media.Id).Url" alt="@Model.Media.AltText" /&gt; <br>}</div>
</div>
</li>
</ul>
</li>
</ul>
<hr>
<h3><strong>Benefits</strong></h3>
<ul>
<li><strong>Improved Usability</strong>: Content editors can manage assets more effectively without leaving the Umbraco platform.</li>
<li><strong>Legal Compliance</strong>: License information ensures assets are used appropriately.</li>
<li><strong>Automatic Fallback</strong>: Expired assets are seamlessly replaced with a predefined fallback image, ensuring a consistent user experience.</li>
<li><strong>Cost-Effective</strong>: Provides basic DAM capabilities without requiring third-party solutions.</li>
</ul>
<hr>
<h3><strong>Optional Enhancements</strong></h3>
<ul>
<li><strong>Global Fallback Image Setting</strong>:
<ul>
<li>Allow a global fallback image to be set once for the entire website.</li>
</ul>
</li>
<li><strong>Expiration Notifications</strong>:
<ul>
<li>Notify editors when assets are nearing expiration.</li>
</ul>
</li>
</ul>
<hr>
<p>By introducing these basic DAM features, Umbraco can empower editors to manage digital assets efficiently while ensuring compliance and maintaining design consistency.</p>

**Comment by skttl** [comment]
You can already do this by adding your required fields to the Image media type.

**Reply by Girish-kl33** [reply]
Thank you skttl,
 If this becomes a core feature, it would make asset management much easier and more efficient for everyone.

**Reply by skttl** [reply]
I don't think the suggested should be in core. It would cause more confusion than good. You can always add what you need in your specific projects. And you can also make it into a package you could simply install.

**Comment by bergmania** [comment]
Hi. Thanks for the proposal.

I'll like to understand why your are not just providing this functionality as a package or in your custom solution?

I doubt all solution would need this.

**Reply by Girish-kl33** [reply]
Hi bergmania, thanks for your feedback.

I built this in a custom project, and it made managing images much easier without relying on third-party tools. That’s why I think it would be great as part of the core—basic asset management features like this can help a lot of users.

For advanced DAM capabilities, third-party packages will still be the best choice. But having these simple features by default makes basic tasks easier and reduces the need for custom solutions.

**Comment by enkelmedia** [comment]
For me, this makes more sense as a package - we have 0 clients that need this - for them, it would only add confusion. The examples also assume that all media are images, a fallback image does not make sense for an audio file or a video.

The additions that you're proposing could be done in less than 15 minutes for a new site that needs it so I would vote for not "bloating" the backoffice with these features for people that might not need it.

BUT 😊 I do see one very valid point here. The media section is more or less a simple DAM (or at least could be made one) so from a communication standpoint Umbraco could promote the fact that it has a "built-in DAM" that is very extendable.

**Comment by mastrup** [comment]
Looking at the proposed changes, I don't see the need for this in the core either. The fields can easily be added manually.

I have looked at integrating our own DAM (Celum) into Umbraco and I think it could be easier. I would need the following to implement it:
1. A custom version of the Media Picker that connects to the DAM.
2. Custom relations to be able to retrieve the pages where a given image is used, to easily locate pages where an item (that is no longer allowed for use) is used.

If it were possible to provide a source other than the media archive in the media picker, we wouldn't need to build our own picker, and whatever was selected in the picker would also be related to the page.
The only thing we would have to worry about is a standardised way for our source to behave, and how the data from the source should look so that Umbraco can handle the items as it normally would for its own media items.

**Reply by jdpnielsen** [reply]
Regarding DAM integrations - i would be careful about implementing the integration as a custom DAM media picker. [QBank](https://marketplace.umbraco.com/package/qbank) is integrated in this way, and its honestly a very bad editor experience. Both because the integration itself could use improvements, but also because it splits the editor workflow between multiple systems.

For a custom [Mux](https://mux.com/) integration we made at [NoA Ignite](https://github.com/NoA-Ignite-dk), we hooked into the filesystem provider and exposed content to the built in media picker & media library that way. This allows us to reuse as much as possible while extending the built in media library.

I would like to see better support for multiple (virtual) filesystems in Umbraco in the future.

**Reply by mastrup** [reply]
I totally agree, hence my suggestion to add it as a source instead of a standalone picker.

Going through the file system provider could also work, but then I guess it would eliminate the media section entirely?
I could see a scenario where you might want to have all your general media served from the DAM, but specific icons, targeted for the website, stored in the Media section.

**Reply by skttl** [reply]
@jdpnielsen that sounds very interesting. Would the normal media picker then be able to select between "Umbraco" media, and Mux files?

**Reply by bergmania** [reply]
I have for a long time wanted to improve the abstraction to the DAM, to make it easier to do stuff like you did for the Mux implementation, but we did not have it prioritised yet.

Ideally also the tree structure from the media section should be abstracted, so it can be read from a different source

**Reply by jdpnielsen** [reply]
> @jdpnielsen that sounds very interesting. Would the normal media picker then be able to select between "Umbraco" media, and Mux files?

Yeah, it acts as a virtual file in the media library and you can move it around etc as normal. We added a specific muxVideo mediatype to differentiate between types.



> I totally agree, hence my suggestion to add it as a source instead of a standalone picker.
> 
> Going through the file system provider could also work, but then I guess it would eliminate the media section entirely? I could see a scenario where you might want to have all your general media served from the DAM, but specific icons, targeted for the website, stored in the Media section.

Yeah, our implementation is a bit hacky which is why its not published on the marketplace.

@bergmania would love if this could get added to the roadmap. If a proper abstraction for package developers is implemented i would be happy to refactor and opensource our Mux integration.

**Reply by skttl** [reply]
Would love to see a demo of that! Even if its hacky.

@bergmania get an RFC going, this could make a huge difference for the CMS as the hub of all kinds of different sources :)
